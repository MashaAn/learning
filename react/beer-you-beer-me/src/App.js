import React, {Component} from 'react';
import './App.css';
import {Input} from "./Input";
import {beerify} from "./BeerOperations";
import {Output} from "./Output";
import {Image} from "./Image";

const originalSong = `
Say you, say me
Say it for always
That's the way it should be
Say you, say me
Say it together
Naturally
I had a dream I had an awesome dream
People in the park playing games in the dark
And what they played was a masquerade
And from behind of walls of doubt a voice was crying out
Say you, say me
Say it for always
That's the way it should be
Say you, say me
Say it together
Naturally
As we go down life's lonesome highway
Seems the hardest thing to do is to find a friend or two
A helping hand - Some one who understands
That when you feel you've lost your way
You've got some one there to say \\"I'll show you\\"
Say you, say me
Say it for always
That's the way it should be
Say you, say me
Say it together
Naturally
So you think you know the answers - Oh no
'Couse the whole world has got you dancing
That's right - I'm telling you
It's time to start believing - Oh yes
Believing who you are: You are a shining star
Say you, say me
Say it for always
That's the way it should be
Say you, say m
Say it together
Naturally
Say it together... naturally
`;


class App extends Component {
    constructor(props) {
        super(props);
        this.state = {sliderValue: 50, songText: originalSong};
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        const sliderValue = event.target.value;
        const songText = beerify(originalSong, sliderValue);
        this.setState({sliderValue: sliderValue, songText: songText})
    }

    render() {
        const beer = <span role="img" aria-label="beer">üçª</span>;
        return (
            <div className="App">
                <header className="App-header">
                    <h1 className="App-title">{beer} Welcome to Reactbeer {beer}</h1>
                </header>
                <div className="container">
                    <Input value={this.state.sliderValue} onChange={this.handleChange}/>
                    <Output songText={this.state.songText}/>
                    <Image/>
                </div>
            </div>
        );
    }
}

export default App;
