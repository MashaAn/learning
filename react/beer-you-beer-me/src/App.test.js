import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {configure, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import seedrandom from "seedrandom";

configure({adapter: new Adapter()});

describe('App.js', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('has expected state', () => {
        const wrapper = shallow(<App/>);
        expect(wrapper.state().sliderValue).toEqual(50);
        expect(wrapper.state().songText).toContain('Beer');
    });
    it('matches snapshot', () => {
        const tree = renderer.create(<App rng={seedrandom('Test')}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});