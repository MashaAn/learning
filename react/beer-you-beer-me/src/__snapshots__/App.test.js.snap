// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      <span
        aria-label="beer"
        role="img"
      >
        üçª
      </span>
       Welcome to Reactbeer 
      <span
        aria-label="beer"
        role="img"
      >
        üçª
      </span>
    </h1>
  </header>
  <div
    className="container"
  >
    <div
      className="Input-sliderContainer"
    >
      <p>
        <input
          className="Input-slider"
          max={100}
          min={0}
          onChange={[Function]}
          type="range"
          value={50}
        />
      </p>
      <p>
        50
      </p>
    </div>
    <div
      className="Output-song"
    >
      <p>
        
        Say Beer, Beer Beer
        Say Beer Beer always
        That'Beer the Beer Beer Beer be
        Say Beer, Beer Beer
        Say Beer together
        Naturally
        Beer had a dream Beer had an awesome dream
        People in the park Beer Beer in the Beer
        And what they Beer Beer a Beer
        And Beer Beer of walls of doubt a voice Beer crying out
        Say Beer, Beer Beer
        Say Beer Beer always
        That'Beer the Beer Beer Beer be
        Say Beer, Beer Beer
        Say Beer together
        Naturally
        As Beer Beer down Beer'Beer Beer highway
        Beer the Beer Beer to Beer is to find a friend Beer two
        Beer Beer hand - Some Beer who understands
        That Beer Beer feel Beer'Beer Beer Beer Beer
        Beer'Beer got some Beer there to Beer "Beer'll Beer Beer"
        Say Beer, Beer Beer
        Say Beer Beer always
        That'Beer the Beer Beer Beer be
        Say Beer, Beer Beer
        Say Beer together
        Naturally
        So Beer Beer Beer Beer the answers - Oh Beer
        'Couse the whole world Beer got Beer dancing
        That'Beer Beer - Beer'Beer Beer Beer
        Beer'Beer time to Beer believing - Oh yes
        Believing who Beer Beer: Beer Beer a Beer star
        Say Beer, Beer Beer
        Say Beer Beer always
        That'Beer the Beer Beer Beer be
        Say Beer, Beer Beer
        Say Beer together
        Naturally
        Say Beer together... Beer
        
      </p>
    </div>
    <img
      alt="Homer and Lionel Richie"
      src="richie.jpg"
    />
  </div>
</div>
`;
